### PR TITLE
Ui Provider

### PR DESCRIPTION
**Pull request by ruslanpalagin**
https://app.clickup.com/t/34abvp5
https://app.clickup.com/t/865bac5jw

### CODE REVIEW

File: contracts_arbitrum/contracts/v3/RewardsDistributor.sol [hunk 1]
```diff
@@ -5,6 +5,7 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts/interfaces/IERC20.sol";
 import "./interfaces/IRewardsDistributor.sol";
 import "./interfaces/IAddressesProvider.sol";
+import "hardhat/console.sol";
```
Comment (boivlad) [inline contracts_arbitrum/contracts/v3/RewardsDistributor.sol:8]: Remove pls

File: contracts_arbitrum/contracts/v3/UiProvider.sol [hunk 2]
```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./PropertyFactory.sol";
+import "./Asset.sol";
+import "hardhat/console.sol";
+import "./interfaces/IRewardsDistributor.sol";
+
+contract UiProvider is OwnableUpgradeable {
```
Comment (boivlad) [inline contracts_arbitrum/contracts/v3/UiProvider.sol:10]: For what OwnableUpgradeable in this contract?

**Review by boivlad — APPROVED** [review]

File: contracts_arbitrum/contracts/v3/SeriesMaster.sol [hunk 3]
```diff
@@ -174,4 +174,12 @@ contract SeriesMaster is ISeriesMaster, ERC721Upgradeable {
   function getSeriesDocuments(uint256 seriesId) external view override returns (Document[] memory) {
     return documents[seriesId];
   }
+
+  function getAsset(uint seriesId) external view returns (ISeriesMaster.Asset memory) {
+    return assets[seriesId];
+  }
+
+  function getDocuments(uint seriesId) external view returns (ISeriesMaster.Document[] memory) {
```
Comment (boivlad) [inline contracts_arbitrum/contracts/v3/SeriesMaster.sol:182]: Same as getSeriesDocuments

File: contracts_arbitrum/hardhat.config.ts [hunk 4]
```diff
@@ -16,21 +16,27 @@ const accounts =
...
     ? [process.env['PRIVATE_KEY']] : { mnemonic: process.env['MNEMONIC'] };
 
+const hardhatNetwork = process.env.NO_FORK ?
+  {
+    deploy: ['deploy/dev']
+  }:
+  {
+    chainId: 421613,
+    forking: {
+      url: 'https://arb-goerli.g.alchemy.com/v2/i7XMrv80YbBZuJBt6u0I7zaAmY9gzERx',
```
Comment (boivlad) [inline contracts_arbitrum/hardhat.config.ts:26]: Alchemy key shode to import from .env

**Review by boivlad — APPROVED** [review]

File: contracts_arbitrum/hardhat.config.ts (hunk 4 above)
Comment (ruslanpalagin) [reply contracts_arbitrum/hardhat.config.ts:26]: it's not me who added this... I did not use fork.
I think the key is public anyway

File: contracts_arbitrum/contracts/v3/SeriesMaster.sol (hunk 3 above)
Comment (ruslanpalagin) [reply contracts_arbitrum/contracts/v3/SeriesMaster.sol:182]: Thanks, I did not noticed it.
Still I think getDocuments is a better name)

File: contracts_arbitrum/contracts/v3/UiProvider.sol (hunk 2 above)
Comment (ruslanpalagin) [reply contracts_arbitrum/contracts/v3/UiProvider.sol:11]: u r right! removing

File: contracts_arbitrum/contracts/v3/RewardsDistributor.sol (hunk 1 above)
Comment (ruslanpalagin) [reply contracts_arbitrum/contracts/v3/RewardsDistributor.sol:8]: +

**Review by boivlad — APPROVED** [review]

**Pull request merged**